tests/unit/helm/app-deployment_test.go

```diff
@@ -222,6 +222,165 @@ func (s *deploymentAppTemplateTest) TestReplicas() {
+						}
+					  }
+					}
+				  ]`
+				err := json.Unmarshal([]byte(eexpectedTopologySpreadConstraintJSON), &expectedTopologySpreadConstraint)
+				s.NoError(err)
+				s.Equal(expectedTopologySpreadConstraint, constraint, "Constraints should be equal")
+			},
+		},
+		{
+			"overrideTopologySpreadConstraintsOptionalValues",
+			map[string]string{
+				"appSettings.topologySpreadConstraints[0].matchLabelKeys":     "[\"pod-template-hash\"]",
+				"appSettings.topologySpreadConstraints[0].maxSkew":            "1",
+				"appSettings.topologySpreadConstraints[0].minDomains":         "1",
+				"appSettings.topologySpreadConstraints[0].nodeAffinityPolicy": "Honor",
+				"appSettings.topologySpreadConstraints[0].nodeTaintsPolicy":   "Honor",
+				"appSettings.topologySpreadConstraints[0].topologyKey":        "kubernetes.io/hostname",
+				"appSettings.topologySpreadConstraints[0].whenUnsatisfiable":  "DoNotSchedule",
+				"appSettings.topologySpreadConstraints[1].matchLabelKeys":     "[\"pod-template-hash\"]",
+				"appSettings.topologySpreadConstraints[1].maxSkew":            "1",
+				"appSettings.topologySpreadConstraints[1].minDomains":         "1",
+				"appSettings.topologySpreadConstraints[1].nodeAffinityPolicy": "Honor",
+				"appSettings.topologySpreadConstraints[1].nodeTaintsPolicy":   "Honor",
+				"appSettings.topologySpreadConstraints[1].topologyKey":        "kubernetes.io/hostname",
+				"appSettings.topologySpreadConstraints[1].whenUnsatisfiable":  "DoNotSchedule",
+			},
+			func(constraint []corev1.TopologySpreadConstraint) {
+				var expectedTopologySpreadConstraint []corev1.TopologySpreadConstraint
+				eexpectedTopologySpreadConstraintJSON := `[
+					{
+					  "matchLabelKeys": [
+					  	"pod-template-hash"
+					  ],
+					  "maxSkew": 1,
+					  "minDomains": 1,
+					  "nodeAffinityPolicy": "Honor",
+					  "nodeTaintsPolicy": "Honor",
+					  "topologyKey": "kubernetes.io/hostname",
+					  "whenUnsatisfiable": "DoNotSchedule",
+					  "labelSelector": {
+					  	"matchLabels": {
+							"app.kubernetes.io/name": "fiftyone-teams-app",
+							"app.kubernetes.io/instance": "fiftyone-test"
+						}
+					  }
+					},
+					{
+					  "matchLabelKeys": [
+					  	"pod-template-hash"
+					  ],
+					  "maxSkew": 1,
+					  "minDomains": 1,
+					  "nodeAffinityPolicy": "Honor",
+					  "nodeTaintsPolicy": "Honor",
+					  "topologyKey": "kubernetes.io/hostname",
+					  "whenUnsatisfiable": "DoNotSchedule",
+					  "labelSelector": {
+					  	"matchLabels": {
+							"app.kubernetes.io/name": "fiftyone-teams-app",
+							"app.kubernetes.io/instance": "fiftyone-test"
+						}
+					  }
+					}
+				  ]`
+				err := json.Unmarshal([]byte(eexpectedTopologySpreadConstraintJSON), &expectedTopologySpreadConstraint)
+				s.NoError(err)
+				s.Equal(expectedTopologySpreadConstraint, constraint, "Constraints should be equal")
+			},
+		},
+		{
+			"overrideTopologySpreadConstraintsSelectorLabels",
```

Comment:
We might have some funky spacing here too? This might/should be indented a bit more as it doesn't align with the above?

Reply:
Is this relating to the `overrideTopologySpreadConstraintsSelectorLabels` line? It looks aligned with the test case name above it and the values below it.

If it relates to the string literal, I changed tabs -> spaces in the literals, so should be resolved.

Reply:
It's all good now.